src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -533,29 +517,28 @@ private void ProcessProjectChange(Solution solution, ProjectId projectId)
             var installedPackages = new MultiDictionary<string, string>();
             var isEnabled = false;
 
-            // Calling into NuGet.  Assume they may fail for any reason.
+            var serviceContainer = _serviceProvider.GetService<IBrokeredServiceContainer, SVsBrokeredServiceContainer>();
```

Comment:
❗ This must be called from the main thread.

Reply:
How come?  Note: running locally, tehre seems to be no issue here. @jasonmalinowski @AArnott  Can service broker only be acquired from the UI thread?

Reply:
Regardless of the service `IServiceProvider.GetService` must always be called from the main thread. There *might* be a couple of *very* special exception services to this rule and Ryan Molden could confirm/refute that if you wondered. But SVsBrokeredServiceContainer is *not* one of them. 
The 99-100% rule is `IServiceProvider` must be accessed from the main thread. The `IAsyncServiceProvider` is free-threaded however, and you can call `GetServiceAsync` from any thread. If the service requires the main thread, it will switch as necessary (following VS threading rules). 
SVsBrokeredServiceContainer is in fact implemented by an async package and async service factory, so you should await GetServiceAsync instead of call GetService anyway to avoid UI delays.

Reply:
Switched to IAsyncServiceProvider.  thanks!